Added BinarySearch Implementation in java and readme file

pranshu1402 opened the pull request.

pranshu1402 committed:
Merge pull request #1 from khanchi97/master

Update from original

pranshu1402 committed:
Adding BinarySearch implementation in java

pranshu1402 committed:
Added readme file for Binary search implementation

khanchi97 reviewed (changes requested):
It's not a good implementation try to use it.Its giving wrong answers.Try to run this
enter the size of the array : 6
enter element 1 : 6
enter element 2 : 4
enter element 3 : 3
enter element 4 : 2
enter element 5 : 1
enter element 6 : 8
Enter element to be searched : 1
Index from iterative binarySearch : -1
Index from recursive binarySearch : -1
How it can give -1 as a result

pranshu1402 commented:
but binary search is used to search in sorted arrays only..??

khanchi97 commented:
So if user enters a value which is not sorted then it is your work to sort the array.

pranshu1402 committed:
Added sorting of input array 

Before Binary search the array must be sorted else it will give undefined results.

22shubh22 reviewed (changes requested):
Binary search of array gives index of array after sorting and not the original array index. Make necessary changes.

pranshu1402 committed:
Merge pull request #2 from khanchi97/master

merging from original

pranshu1402 committed:
Added LIS dp problem in java

pranshu1402 committed:
Added Kadane dp algo in java

pranshu1402 committed:
Added gcd euclid algo in java

pranshu1402 committed:
Updated binary search  java as requested

Removed sorting of arrays after taking input.v because its a binary search it works only on sorted arrays. Its of no use on other arrays as it will then affect its time complexity as well as results. So asking user initially to enter the array only in increasing order.

22shubh22 reviewed (changes requested):
error in running binarysearch.java

pranshu1402 commented:
how to remove a file from commit?

pranshu1402 commented:
actually that implementation has already been merged before

22shubh22 commented:
@khanchi97 please help.

khanchi97 commented:
@pranshu1402 if you want to change some files in your previous commits then take a look at this command `git reset --soft commit_hash` and google about this what it does and then you'll have a better idea, how to change files in your previous commits.
Actually what this command does is unstage all the files before that commit_hash then you can `git add` only those files which you want to commit.
